From 5bb2576195ce37cae751a6226d18339684eb19e5 Mon Sep 17 00:00:00 2001
From: Will Dietz <w@wdtz.org>
Date: Mon, 28 Oct 2019 13:55:46 -0500
Subject: [PATCH] ell/tls: fix crash, don't index through adj. stack objs

Building with clang may make this more likely to crash,
problem is encountered on ell/tls.c:114 (before).
---
 ell/tls.c | 5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/ell/tls.c b/ell/tls.c
index 0e06c27..a947ae4 100644
--- a/ell/tls.c
+++ b/ell/tls.c
@@ -92,7 +92,8 @@ bool tls12_prf(enum l_checksum_type type,
 {
 	struct l_checksum *hmac = l_checksum_new_hmac(type, secret, secret_len);
 	size_t a_len, chunk_len, prfseed_len = strlen(label) + seed_len;
-	uint8_t a[128], prfseed[prfseed_len];
+	uint8_t a[128 + prfseed_len];
+	uint8_t *prfseed = &a[128];
 
 	if (!hmac)
 		return false;
@@ -108,7 +109,7 @@ bool tls12_prf(enum l_checksum_type type,
 		/* Generate A(i) */
 		l_checksum_reset(hmac);
 		l_checksum_update(hmac, a, a_len);
-		a_len = l_checksum_get_digest(hmac, a, sizeof(a));
+		a_len = l_checksum_get_digest(hmac, a, 128);
 
 		/* Append seed & generate output */
 		memcpy(a + a_len, prfseed, prfseed_len);
-- 
2.24.0-rc1

